src/test/lib/ApiTest.cpp

```diff
@@ -1813,10 +2113,151 @@ void QuicTestGlobalSetParam()
             QUIC_STATUS_INVALID_PARAMETER,
             MsQuic->SetParam(
                 nullptr,
-                QUIC_PARAM_GLOBAL_LIBRARY_VERSION,
+                QUIC_PARAM_GLOBAL_LIBRARY_GIT_HASH,
+                0,
+                nullptr));
+    }
+
+    //
+    // QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS
+    //
+    {
+        TestScopeLogger LogScope("QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS");
+        //
+        // BufferLength is not divisible by sizeof(uint16_t)
+        //
+        {
+            TestScopeLogger LogScope("BufferLength is not divisible by sizeof(uint16_t)");
+            uint16_t Data[4];
+            TEST_QUIC_STATUS(
+                QUIC_STATUS_INVALID_PARAMETER,
+                MsQuic->SetParam(
+                    nullptr,
+                    QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS,
+                    sizeof(Data) + 1,
+                    &Data));
+        }
+
+        //
+        // one of data is bigger than the number of its' platform cpus
+        //
+        {
+            TestScopeLogger LogScope("one of data is bigger than the number of its' platform cpus");
+            uint16_t Data[4] = {};
+            Data[0] = UINT16_MAX;
+            TEST_QUIC_STATUS(
+                QUIC_STATUS_INVALID_PARAMETER,
+                MsQuic->SetParam(
+                    nullptr,
+                    QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS,
+                    sizeof(Data),
+                    &Data));
+        }
+
+        //
+        // Good without data
+        //
+        {
+            TestScopeLogger LogScope("Good without data");
+            TEST_QUIC_SUCCEEDED(
+                MsQuic->SetParam(
+                    nullptr,
+                    QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS,
+                    0,
+                    nullptr));
+        }
+
+        //
+        // Good with data
+        //
+        {
+            TestScopeLogger LogScope("Good with data");
+            uint16_t Data[4] = {};
+            TEST_QUIC_SUCCEEDED(
+                MsQuic->SetParam(
+                    nullptr,
+                    QUIC_PARAM_GLOBAL_DATAPATH_PROCESSORS,
+                    sizeof(Data),
+                    &Data));
+        }
+    }
+
+#if QUIC_TEST_DATAPATH_HOOKS_ENABLED
```

Comment:
I don't think you should add tests for test hooks. They already get used.

Reply:
Fixed, thanks!